closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityConfigurator.java

```diff
@@ -0,0 +1,325 @@
+
+        // JDK properties name as Key, restricted security properties vaule as value.
+        propsMapping.put("jdk.tls.disabledNamedCurves", properties.getJdkTlsDisabledNamedCurves());
+        propsMapping.put("jdk.tls.disabledAlgorithms", properties.getJdkTlsDisabledAlgorithms());
+        propsMapping.put("jdk.tls.ephemeralDHKeySize", properties.getJdkTlsDphemeralDHKeySize());
+        propsMapping.put("jdk.tls.legacyAlgorithms", properties.getJdkTlsLegacyAlgorithms());
+        propsMapping.put("jdk.certpath.disabledAlgorithms", properties.getJdkCertpathDisabledAlgorithms());
+        propsMapping.put("jdk.security.legacyAlgorithm", properties.getJdkSecurityLegacyAlgorithm());
+
+        for (Map.Entry<String, String> entry : propsMapping.entrySet()) {
+            String jdkPropsName = entry.getKey();
+            String propsNewValue = entry.getValue();
+
+            String propsOldValue = !isNullOrBlank(props.getProperty(jdkPropsName)) ? props.getProperty(jdkPropsName) : "";
+
+            if (!isNullOrBlank(propsNewValue)) {
+                String values = !isNullOrBlank(propsOldValue) ? propsOldValue + ", " + propsNewValue : propsNewValue;
+                props.setProperty(jdkPropsName, values);
+                if (debug != null) {
+                    debug.println("Added restricted security properties, with property: " + jdkPropsName + " values: "
+                            + values);
+                }
+            }
+        }
+
+        // For keyStore and keystore.type, old value not needed, just set the new value.
+        if (!isNullOrBlank(properties.getKeyStoreType())) {
+            props.setProperty("keystore.type", properties.getKeyStoreType());
+        }
+        if (!isNullOrBlank(properties.getKeyStore())) {
+            System.setProperty("javax.net.ssl.keyStore", properties.getKeyStore());
+        }
+    }
+
+    /**
+     * Check if restricted security policy sunset.
+     *
+     * @param descSunsetDate the sun set date from java.security
+     * @return true if the restricted security policy sunset
+     */
+    private static boolean isPolicySunset(String descSunsetDate) {
+
+        boolean isSunset = false;
+        try {
+            isSunset = LocalDate.parse(descSunsetDate, DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                    .isBefore(LocalDate.now());
+        } catch (Exception except) {
+            new RuntimeException(
+                    "Restricted security policy sunset date is inccorect, the correct format is yyyy-MM-dd.")
+                    .printStackTrace();
+            System.exit(1);
+        }
+
+        if (debug != null) {
+            debug.println("Restricted security policy is sunset: " + isSunset);
+        }
+        return isSunset;
+    }
+
+    /**
+     * Check if the input string is null and empty.
+     *
+     * @param string the input string
+     * @return true if the input string is null and emtpy
```

Comment:
The text should say "string is null **or blank**" not "string is null **and empty**".